Code/src/main/java/nl/utwente/group10/ui/components/anchors/ConnectionAnchor.java

```diff
@@ -48,55 +48,115 @@ public ConnectionAnchor(Block block, CustomUIPane pane) {
     public abstract Type getType();
 
     /**
-     * Disconnects the anchor from the connection
+     * Removes the connection from its pane, first disconnecting it from its
+     * anchors.
      *
      * @param connection
-     *            Connection to disconnect from.
+     *            Connection to remove from its pane.
      */
-    public abstract void removeConnection(Connection connection);
+    public void removeConnection(Connection connection) {
+        if (connections.contains(connection)) {
+            connections.remove(connection);
+            connection.remove();
+        }
+    }
 
     /**
-     * Set the connection this anchor is connected to.
+     * Disconnects the connection from this anchor, keeping the connection on
+     * its pane.
      *
      * @param connection
+     *            Connection to disconnect from.
+     */
+    public void disconnectConnection(Connection connection) {
+        if (connections.contains(connection)) {
+            connections.remove(connection);
+            connection.disconnect(this);
+        }
+    }
+
+    /**
+     * Removes all the connections this anchor has.
+     */
+    public void removeConnections() {
+        while (connections.size() > 0) {
+            removeConnection(connections.get(0));
+        }
+    }
+
+    /**
+     * Disconnects all the connections this anchor has from this anchor.
+     */
+    public void disconnectConnections() {
+        while (connections.size() > 0) {
+            disconnectConnection(connections.get(0));
+        }
+    }
+
+    /**
+     * Adds the given connection to the connections this anchor has.
+     * 
+     * @param connection
+     *            Connection to add
      */
-    public void setConnection(Connection connection) {
-        this.connection = Optional.ofNullable(connection);
+    public void addConnection(Connection connection) {
+        connections.add(connection);
     }
 
-    /** Returns true if the anchor is connected to a connection. */
+    /** Returns true this anchor has 1 or more connections. */
     public boolean hasConnection() {
-        return connection.isPresent();
+        return connections.size() > 0;
     }
 
     /**
-     * @return True if this ConnectionAnchor is connected to a Connection and
-     *         that connection is fully connected.
+     * @return True if the primary connection is connected.
+     */
+    public boolean isPrimaryConnected() {
+        return isConnected(0);
+    }
+    
+    /**
+     * @param index Index of the connection to check
+     * @return Wether or not the connection specified by the index is connected.
      */
-    public boolean isConnected() {
-        return hasConnection() && getConnection().get().isConnected();
+    public boolean isConnected(int index){
+        return index >= 0 && index < getConnections().size() && getConnections().get(index).isConnected();
```

Comment:
Hmm, klopt. Als je dat specifiek wilt zou ik dat even opnemen in de `@return` van de Javadoc.